deployment/ccip/changeset/cs_deploy_chain.go

```diff
@@ -560,15 +556,37 @@ func solRouterProgramData(e deployment.Environment, chain deployment.SolChain, c
 	return programData, nil
 }
 
-func checkRouterInitialized(e deployment.Environment, chain deployment.SolChain, ccipRouterProgram solana.PublicKey) (bool, error) {
-	routerConfigPDA := GetRouterConfigPDA(ccipRouterProgram)
-	routerConfigInfo, err := chain.Client.GetAccountInfoWithOpts(e.GetContext(), routerConfigPDA, &solRpc.GetAccountInfoOpts{
-		Commitment: solRpc.CommitmentConfirmed,
-	})
+func initialzeRouter(e deployment.Environment, chain deployment.SolChain, ccipRouterProgram solana.PublicKey) error {
+	programData, err := solRouterProgramData(e, chain, ccipRouterProgram)
 	if err != nil {
-		return false, nil
+		return fmt.Errorf("failed to get solana router program data: %w", err)
 	}
-	return routerConfigInfo != nil && len(routerConfigInfo.Value.Data.GetBinary()) > 0, nil
+
+	instruction, err := solRouter.NewInitializeInstruction(
+		chain.Selector,                // chain selector
+		deployment.SolDefaultGasLimit, // default gas limit
+		true,                          // allow out of order execution
+		EnableExecutionAfter,          // period to wait before allowing manual execution
+		solana.PublicKey{},
+		GetRouterConfigPDA(ccipRouterProgram),
+		GetRouterStatePDA(ccipRouterProgram),
+		chain.DeployerKey.PublicKey(),
+		solana.SystemProgramID,
+		ccipRouterProgram,
+		programData.Address,
+		GetExternalExecutionConfigPDA(ccipRouterProgram),
+		GetExternalTokenPoolsSignerPDA(ccipRouterProgram),
+	).ValidateAndBuild()
+
+	if err != nil {
+		return fmt.Errorf("failed to build instruction: %w", err)
+	}
+	err = chain.Confirm([]solana.Instruction{instruction})
+
+	if err != nil {
+		return fmt.Errorf("failed to confirm instructions: %w", err)
+	}
```

Comment:
```suggestion
	).ValidateAndBuild()
	if err != nil {
		return fmt.Errorf("failed to build instruction: %w", err)
	}

	err = chain.Confirm([]solana.Instruction{instruction})
	if err != nil {
		return fmt.Errorf("failed to confirm instructions: %w", err)
	}
```
nit: `if err != nil` should hug the `err = ...` assignment where possible.